modules/dmrpp_module/build_dmrpp_util.cc

```diff
@@ -1867,22 +1869,48 @@ static string recreate_cmdln_from_args(int argc, char *argv[])
     return ss.str();
 }
 
+/**
+ * @brief Returns an ISO-8601 date time string for the time at which this function is called.
+ * Tip-o-the-hat to Morris Day and The Time...
+ * @return An ISO-8601 date time string
+ */
+std::string what_time_is_it(){
+    // Get current time as a time_point
+    auto now = std::chrono::system_clock::now();
+
+    // Convert to system time (time_t)
+    auto time_t_now = std::chrono::system_clock::to_time_t(now);
+
+    // Convert to tm structure (GMT time)
+    struct tm tbuf{};
+    const std::tm* gmt_time = gmtime_r(&time_t_now, &tbuf);
+
+    // Format the time using a stringstream
+    std::stringstream ss;
```

Comment:
Oy, stringstream... In C++17 you can write directly to a std::string. Unfortunately, we're at 14 and stringstream is slower by more than order of magnitude than writing the chars to a vector and returning that as a string. I don't have a great solution for that without looking it up.

Reply:
I lieu of string stream we could brute force it:
```
    string iso_8601;
    iso_8601.append(to_string(gmt_time->tm_year)).append("-");
    iso_8601.append(to_string(gmt_time->tm_mon)).append("-");
    iso_8601.append(to_string(gmt_time->tm_mday)).append("T");
    iso_8601.append(to_string(gmt_time->tm_hour)).append(":");
    iso_8601.append(to_string(gmt_time->tm_min)).append(":");
    iso_8601.append(to_string(gmt_time->tm_sec)).append("Z");
```

Reply:
Actually, I looked around in the BES and elsewhere we do it using `strftime`:
```
    auto iso8601_fmt = "%Y-%m-%dT%H:%M:%SZ";
    char time_buf[strlen(iso8601_fmt)];
    std::strftime(time_buf, sizeof buf, iso8601_fmt, gmt_time);
```